### ISSUE TITLE
GlobalDependencyContext inject only on root gameobject of new loaded scene

### ISSUE THREAD

**Issue by viveleltsi** [juniordiscart/ImpossibleOdds-Toolkit#1]
# Description
When a new scene is loaded the global context is injected only on root gameobject.

From doc : `Now that global resources have been installed, each time a scene is loaded, it will scan that scene and inject any component that requires its resources.` As I understand it is injected to any component that requires. There's no mention of hierarchy position. 

# Expected
It should be injected to any gameobject even none root one **or** the documentation should be updated accordly.

**Comment by juniordiscart** [comment]
Hi

Thanks a ton for the report! I looked it up and there's a parameter missing in the `OnSceneLoaded` method of `GlobalDependencyContext`. The `Inject` method should have an added `true` value that will inject the children as well.

```cs
private void OnSceneLoaded(Scene scene, LoadSceneMode sceneLoadMode)
{
	Log.Info("Injecting scene '{0}' with the global dependency scope.", scene.name);
	scene.GetRootGameObjects().Inject(DependencyContainer, true); // <- True to include children
}
```

I'll create a fix for a new release.
Thanks again! :)

**Comment by viveleltsi** [comment]
Hello

Yes it what I thought. I made a dirty fix for my project for now by creating a new file (to not modify your library).
```
    public class GloablDependencyContextFix
	{

		private static GloablDependencyContextFix globalContext = null;


		private GloablDependencyContextFix()
		{
			SceneManager.sceneLoaded += OnSceneLoaded;
		}

		[RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSceneLoad)]
		private static void InitializeGlobalContext()
		{
			globalContext = new GloablDependencyContextFix();
		}

		private void OnSceneLoaded(UnityEngine.SceneManagement.Scene scene, LoadSceneMode sceneLoadMode)
		{
			Log.Info("Injecting scene '{0}' with the global dependency context.", scene.name);
			scene.GetRootGameObjects().Inject(GlobalDependencyContext.GlobalContext.DependencyContainer, true );
		}
	}
```

Thank you for your quick answer and for your tools!

**Comment by juniordiscart** [comment]
This should be fixed now in the v1.0.2 release. :)
